### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by mish-sk**

### CODE REVIEW

**Comment by mish-sk** [comment]
Знімок екрана 2023-02-19 о 14 36 44
Знімок екрана 2023-02-19 о 14 36 53
Знімок екрана 2023-02-19 о 14 37 35
Знімок екрана 2023-02-19 о 14 37 45
Знімок екрана 2023-02-19 о 14 40 15
Знімок екрана 2023-02-19 о 14 40 23

File: blog/models.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+class Commentary(models.Model):
+    user = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
+        on_delete=models.CASCADE
+    )
+    post = models.ForeignKey(
+        Post,
+        on_delete=models.CASCADE,
+        related_name="comment"
```
Comment (olena-hul) [inline blog/models.py:31]: The post can have many comments, change the related name to plural

File: blog/views.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        comment = Commentary.objects.create(
+            user=request.user,
+            post_id=pk,
+            content=content
+        )
+        comment.save()
+
+    context = {
+        "post": post,
+        "commentaries": commentaries,
```
Comment (olena-hul) [inline blog/views.py:32]: you can make post.comments, no need to pass additional value into context

File: blog/views.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+class PostListView(generic.ListView):
+    model = Post
+    context_object_name = "post_list"
+    template_name = "blog/index.html"
+    paginate_by = 5
+
+
+def post_detail_view(request, pk):
+    post = get_object_or_404(
+        Post.objects.select_related("owner").prefetch_related("comment__user"),
```
Comment (olena-hul) [inline blog/views.py:16]: Add prefetch related for comments too

File: blog/views.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        pk=pk
+    )
+    commentaries = Commentary.objects.filter(post_id=pk)
```
Comment (olena-hul) [inline blog/views.py:19]: No need for this query

File: blog/views.py (hunk 2 above)
Comment (olena-hul) [reply blog/views.py:32]: Indeed, you never use it in templates

**Review by olena-hul — CHANGES_REQUESTED** [review]

**Review by olena-hul — APPROVED** [review]

**Comment by olena-hul** [comment]
Good job!

**Pull request closed without merging**